### PR TITLE
Send foreground status to the host application and handle extensible Logcat requests

### PR DESCRIPTION
**Pull request by jmatsu**
- Send foreground/background events to the host application
  - Don't use any lifecycle API from AndroidX but ActivityLifecycleCallback is enough for this use-case.
  - Foreground events may be sent before a valid service connection has been established so buffer and flush are required.
- Allow to use a capture id from Logcat request

### CODE REVIEW

File: sdk/src/main/java/com/deploygate/sdk/DeployGate.java [hunk 1]
```diff
@@ -332,6 +392,19 @@ private void invokeAction(
...
     }
 
+    private void flushPendingEvents() {
+        // cannot re-enqueue events for now
+        synchronized (sPendingEventLock) {
+            Iterator<Map.Entry<String, Bundle>> iterator = mPendingEvents.entrySet().iterator();
+
+            while (iterator.hasNext()) {
+                Map.Entry<String, Bundle> entry = iterator.next();
+                invokeAction(entry.getKey(), entry.getValue(), false);
```
Comment (jmatsu) [inline sdk/src/main/java/com/deploygate/sdk/DeployGate.java:402]: The current buffer mechanism is relaxed. Enough.

**Comment by satsukies** [comment]
@jmatsu can I begin review this?

File: sdk/src/main/java/com/deploygate/sdk/LogcatInstructionSerializer.java [hunk 2]
```diff
@@ -183,6 +191,7 @@ int sendSingleChunk(
         Bundle extras = request.toExtras();
 
         try {
+
```
Comment (satsukies) [inline sdk/src/main/java/com/deploygate/sdk/LogcatInstructionSerializer.java:194]:
```suggestion
```

**Comment by jmatsu** [comment]
@satsukies Yes, please review the both of this PR and another one.

**Review by satsukies — APPROVED** [review]
VisibilityLifecycleCallbacks is amazing 👍 LGTM

**Comment by jmatsu** [comment]
Thank you for your reviews.

**Pull request merged**